node/node.go

```diff
@@ -378,8 +378,14 @@ func (nc *Config) Build(ctx context.Context) (*Node, error) {
 	// set up pinger
 	pingService := ping.NewPingService(peerHost)
 
+	// DO NOT MERGE this code smells like poo, this smell exists elsewhere,
+	// should have defaults set on it _before_ we get here?
+	if nc.Clock == nil {
```

Comment:
I'm not a fan of this pattern, however since `--block-time` is a command line option I am unsure if there is a way around it. If there are no suggestions during review I will remove this smelly comment.

Reply:
What about passing an integer parsed from the CLI (or the default) on the node config and passing that to a clock constructor during the `porcelain.New` call below?

Reply:
You leave the concrete implementation of `BlockClock` configurable here for testing?  Like for a FAST test or something?  If not I kind of agree with @ZenGround0.. you can just instantiate the `BlockClock` given the `block-time` option.